Libraries/Azure.psm1

```diff
@@ -2397,23 +2397,73 @@ function Add-AzureAccountFromSecretsFile {
 
     if ( Test-Path $secretsFile ) {
         Write-LogInfo "$secretsFile found."
-        Write-LogInfo "------------------------------------------------------------------"
-        Write-LogInfo "Authenticating Azure PS session.."
         $XmlSecrets = [xml](Get-Content $secretsFile)
         $ClientID = $XmlSecrets.secrets.SubscriptionServicePrincipalClientID
         $TenantID = $XmlSecrets.secrets.SubscriptionServicePrincipalTenantID
         $Key = $XmlSecrets.secrets.SubscriptionServicePrincipalKey
-        $pass = ConvertTo-SecureString $key -AsPlainText -Force
-        $mycred = New-Object System.Management.Automation.PSCredential ($ClientID, $pass)
+        $AzureContextFilePath = $XmlSecrets.secrets.AzureContextFilePath
         $subIDSplitted = ($XmlSecrets.secrets.SubscriptionID).Split("-")
         $subIDMasked = "$($subIDSplitted[0])-xxxx-xxxx-xxxx-$($subIDSplitted[4])"
-
-        $null = Add-AzureRmAccount -ServicePrincipal -Tenant $TenantID -Credential $mycred
-        $selectedSubscription = Select-AzureRmSubscription -SubscriptionId $XmlSecrets.secrets.SubscriptionID
+        Write-LogInfo "------------------------------------------------------------------"
+        if ($ClientID -and $Key) {
+            Write-LogInfo "Authenticating Azure PS session using Service Principal..."
+            $pass = ConvertTo-SecureString $key -AsPlainText -Force
+            $mycred = New-Object System.Management.Automation.PSCredential ($ClientID, $pass)
+            $null = Add-AzureRmAccount -ServicePrincipal -Tenant $TenantID -Credential $mycred
+        } elseif ($AzureContextFilePath) {
+            Write-LogInfo "Authenticating Azure PS session using saved context file $AzureContextFilePath..."
+            $null = Import-AzureRmContext -Path $AzureContextFilePath
+        } else {
+            $ContextFiles = Get-ChildItem -Path $PWD -Recurse | `
+                Where-Object { $_.Name.EndsWith(".json") } | Select-String -Pattern "AzureCloud" | Select Path
+            $ContextFiles = $ContextFiles.Path | Get-Unique
+            if ($ContextFiles.Count -eq 0) {
```

Comment:
what if the file has information, but not enough or correct, or even meaningless? The line 2420 needs revision.

Reply:
@juhlee-microsoft If the file information is not correct then the import command will fail. Which will result in failure of 'Select-AzureRmSubscription" command which is handled by LISAv2 and will exit with 1, as expected.